tests/test_transforms.py

```diff
@@ -172,18 +170,252 @@ def test_transformed_logprob(at_dist, dist_params, sp_dist, size):
+
+    y_rv = at.random.normal(0, 1, name="y_rv", size=rv_size) * scale
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([scale, y], logp)
+
+    scale_test_val = 4.0 if scalar_scale else np.full(rv_size, 4.0)
+    y_test_val = np.full(rv_size, 1.0)
+
+    assert np.isclose(
+        logp_fn(scale_test_val, y_test_val),
+        sp.stats.norm(0, scale_test_val).logpdf(y_test_val).sum(),
+    )
+
+
+@pytest.mark.parametrize("rv_size", [None, 1, 2, (3, 2)])
+def test_affine_transform(rv_size):
+    loc = at.scalar("loc")
+
+    if rv_size is None:
+        scale = at.scalar("scale")
+    elif isinstance(rv_size, int):
+        scale = at.vector("scale")
+    else:
+        scale = at.matrix("scale")
+
+    y_rv = loc + at.random.normal(0, 1, name="y_rv", size=rv_size) * scale
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([loc, scale, y], logp)
+
+    loc_test_val = 4.0
+    scale_test_val = np.full(rv_size, 0.5)
+    y_test_val = np.full(rv_size, 1.0)
+
+    assert np.isclose(
+        logp_fn(loc_test_val, scale_test_val, y_test_val),
+        sp.stats.norm(loc_test_val, scale_test_val).logpdf(y_test_val).sum(),
+    )
+
+
+def test_double_scalar_loc_transform():
+    loc1 = at.scalar("loc1")
+    loc2 = at.scalar("loc2")
+    y_rv = loc1 + loc2 + at.random.normal(0, 1, name="y_rv")
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y})
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([loc1, loc2, y], logp)
+
+    loc1_test_val = 2.0
+    loc2_test_val = 2.0
+    y_test_val = 1.0
+
+    assert np.isclose(
+        logp_fn(loc1_test_val, loc1_test_val, y_test_val),
+        sp.stats.norm(loc1_test_val + loc2_test_val, 1).logpdf(y_test_val),
+    )
+
+
+def test_double_scalar_scale_transform():
+    scale1 = at.scalar("scale1")
+    scale2 = at.scalar("scale2")
+    y_rv = at.random.normal(0, 1, name="y_rv") * scale1 * scale2
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y})
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([scale1, scale2, y], logp)
+
+    scale1_test_val = 2.0
+    scale2_test_val = 2.0
+    y_test_val = 1.0
+
+    assert np.isclose(
+        logp_fn(scale1_test_val, scale1_test_val, y_test_val),
+        sp.stats.norm(0, scale1_test_val * scale2_test_val).logpdf(y_test_val),
+    )
+
+
+def test_exp_scale_transform():
+    b = at.scalar("b")
+    base_rv = at.random.normal(0, 1, name="base_rv", size=2)
+    y_rv = at.exp(base_rv * b)
+    y_rv.name = "y_rv"
+
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y})
+    logp_fn = aesara.function([b, y], logp)
+
+    b_val = 1.5
+    y_val = [0.1, 0.1]
+
+    assert np.allclose(logp_fn(b_val, y_val), sp.stats.lognorm(b_val).logpdf(y_val))
+
+
+def test_affine_log_transform():
```

Comment:
This tests passes with `size=None`, and it seems related to the ChainedTransform `log_jac_det` returning a wrong value for non-scalar shapes...